src/transformers/pipelines/zero_shot_classification.py

```diff
@@ -65,24 +65,42 @@ class ZeroShotClassificationPipeline(Pipeline):
     def __init__(self, args_parser=ZeroShotClassificationArgumentHandler(), *args, **kwargs):
         super().__init__(*args, **kwargs)
         self._args_parser = args_parser
-        if self.entailment_id == -1:
+        if self.model.config.model_type == "t5":
+            self._get_entailment_neutral_contradiction_token_id()
+        elif self.entailment_id == -1:
             logger.warning(
                 "Failed to determine 'entailment' label id from the label2id mapping in the model config. Setting to "
                 "-1. Define a descriptive label2id mapping in the model config to ensure correct outputs."
             )
 
     @property
     def entailment_id(self):
+        if self.model.config.model_type == "t5":
+            return -1
         for label, ind in self.model.config.label2id.items():
             if label.lower().startswith("entail"):
                 return ind
         return -1
 
+    def _get_entailment_neutral_contradiction_token_id(self):
+        class_ids = self.tokenizer(["entailment", "neutral", "contradiction"]).input_ids
+        self.entailment_token_id, self.neutral_token_id, self.contradiction_token_id = [ids[0] for ids in class_ids]
```

Comment:
You cannot assume that `ids[0]` actually corresponds to these words. This needs to be at the very least checked (if a word is chunked, results are probably going to be bad, what about byte level tokenizers (256 byte tokens) ?

Discarding any tokenizer that does not work is acceptable.

Reply:
As I focused just on the T5 I assumed that this simplification will work well. With the T5 tokenizer the class names are tokenized as:

- entailment: ["", "en", "tail", "ment"]
- neutral: ["neutral"]
- contradiction: ["contradiction"]

There is no problem with neutral and contradiction, but the entailment string is splited into 4 subtokens. My intuition was that if the model generates the first token of entailment, the probability of generating the rest of the tokens after that should be very high.  Therefore, we could estimate the probability of generating the whole entailment string just with the probability of the first token.  Doing it in this way, we can simplify a lot the inference, but it is true that we should measure how it affects the quality of the model.